src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java

```diff
@@ -145,7 +145,14 @@ public AxisIterator iterateAxis(byte axisNumber, NodeTest nodeTest) {
      */
     @Override
     public int compareOrder(NodeInfo nodeInfo) {
-        return getLocalPart().compareTo(nodeInfo.getLocalPart());
+        int cmp = getLineNumber() - nodeInfo.getLineNumber();
```

Comment:
Rename it `result` to fix spelling failure

Reply:
done